### PR TITLE
ACAS-343: Run acasclient tests against appropriate version of ACAS (PR for release/2022.1.x)

### PR DESCRIPTION
**Pull request by bffrost**
## Description
Modify acasclient GitHub Actions to run against correlated ACAS branch. This is part of a series of 4 PRs to update both release branches of both the `acasclient` and `acas` repos.
- For commits to `main` and `release/*` branches (i.e. `release/1.13.7`, `release/2022.1.x`), run tests against the corresponding ACAS branch by the same name.
- For pull requests, run acasclient tests against the target branch, i.e. ACAS `release/2022.1.x` if the acasclient PR is targeting `release/2022.1.x`

## How Has This Been Tested?
The GitHub action passed and I confirmed it ran `test_004_delete_lot` which is a newer, 2022.1.x only test.

### CODE REVIEW

File: .github/workflows/run_tests.yml
Comment (brianbolt) [inline .github/workflows/run_tests.yml:42]: Instead of hardcoding master here you could do:

```
curl -s https://api.github.com/repos/mcneilco/acas | jq -r '.default_branch'
```

**Review by brianbolt — APPROVED** [review]
LGTM.  Approving but made a easy suggestion to get the default branch of acas from the api instead of hardcoding master.

File: .github/workflows/run_tests.yml
Comment (brianbolt) [inline .github/workflows/run_tests.yml:5]: I just found a nice feature which is to use: $default-branch macro:
Example:
https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml#L5

https://github.blog/changelog/2020-07-22-github-actions-better-support-for-alternative-default-branch-names/

**Review by brianbolt — APPROVED** [review]
My second review is still LGTM. But I think we should add this $default-branch macro (see my comments)

File: .github/workflows/run_tests.yml
Comment (brianbolt) [reply .github/workflows/run_tests.yml:5]: Ignore this comment.  I guess the $default-branch macro is for action templates, not for the actual GitHub actions yaml:

https://github.com/github/docs/discussions/5071

**Review by brianbolt — CHANGES_REQUESTED** [review]
Sorry, I keep changing my mind on this.  Because these changes are also occurring on 1.13.7 branch (which is the downstream release). I think we need to up merge those commits into release/2022.1.x rather than making these new commits.

File: .github/workflows/run_tests.yml
Comment (bffrost) [reply .github/workflows/run_tests.yml:42]: Hm.... do we actually want the default branch though? I think we already cover testing against the latest release branch via the corresponding release branch of acasclient.

My unspoken assumption here is that we'd upmerge `release/1.13.7` -> `release/2022.1.x` -> `master` in acas, and `release/1.13.7` -> `release/2022.1.x` -> `main` in acasclient, so I was hardcoding a correspondence between `master` and `main`

**Comment by bffrost** [comment]
Makes sense. I'm closing this PR in favor of #76 which goes into `release/1.13.7`. Then once that's merged I'll go ahead with the upmerging you describe.

**Pull request closed without merging**
